### PR TITLE
R4R: Replace circleci with github actions

### PR DESCRIPTION
**Pull request by oncloudit**

### CODE REVIEW

File: .github/workflows/buildAndPush.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: BuildAndPush
```
Comment (zhangyelong) [inline .github/workflows/buildAndPush.yml]: `BuildAndPublish` is better, `push` is confused with the `git push`

File: .github/workflows/buildAndPush.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+on:
+  pull_request:
```
Comment (zhangyelong) [inline .github/workflows/buildAndPush.yml]: Does this mean that `build` and `publish` will be triggered when PR to the master branch? If so, this should be removed, PRs should not been builded or published until they are merged.

File: .github/workflows/buildAndPush.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        run: source ~/.bashrc; sudo docker build -t irisnet/irishub:testnet-`cat ${{github.sha}}.TAG` --build-arg NetworkType=testnet .
+
+#   push-images:
+#     name: Push Images
+#     runs-on: self-hosted
+#     needs: build-images
+#     steps:
+      - name: Login dockerhub
+        run: sudo docker login -u ${{secrets.DOCKER_USER}} -p ${{secrets.DOCKER_PASS}}
+      - name: push irishub testnet image
```
Comment (zhangyelong) [inline .github/workflows/buildAndPush.yml]:
```suggestion
      - name: Publish irishub testnet image
```

File: .github/workflows/buildAndPush.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        run: sudo docker push irisnet/irishub:testnet-`cat ${{github.sha}}.TAG`
+      - name: push irishub mainnet image
```
Comment (zhangyelong) [inline .github/workflows/buildAndPush.yml]:
```suggestion
      - name: Publish irishub mainnet image
```

File: .github/workflows/buildAndPush.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+      - name: Get image tag      
+        run: |
+          if [[ ${{github.ref}} == "refs/heads/master" ]]; then
+            echo latest > ${{github.sha}}.TAG;
+          else
+            echo ${${{github.ref}}##/} > ${{github.sha}}.TAG;
+          fi
+      - name: build irishub mainnet image
```
Comment (zhangyelong) [inline .github/workflows/buildAndPush.yml]:
```suggestion
      - name: Build irishub mainnet image
```

File: .github/workflows/buildAndPush.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        run: source ~/.bashrc; sudo docker build -t irisnet/irishub:`cat ${{github.sha}}.TAG` .
+      - name: build irishub testnet image
```
Comment (zhangyelong) [inline .github/workflows/buildAndPush.yml]:
```suggestion
      - name: Build irishub testnet image
```

File: .github/workflows/buildAndPush.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    tags:
+      - v.*
+
+jobs:
+  build-images:
+    name: Build Images
+#    runs-on: self-hosted
+    runs-on: ubuntu-16.04
+    steps:
+      - name: Check out code into the Go module directory
```
Comment (zhangyelong) [inline .github/workflows/buildAndPush.yml]:
```suggestion
      - name: Checkout source code
```

File: .github/workflows/testUnit.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+name: TestUnit
+on: [push, pull_request]
+
+jobs:
+  test-images:
+    name: Test Images
+#    runs-on: self-hosted
+    runs-on: ubuntu-16.04
+    steps:
+      - name: Check out code into the Go module directory
```
Comment (zhangyelong) [inline .github/workflows/testUnit.yml]:
```suggestion
      - name: Checkout source code
```

**Comment by oncloudit** [comment]
commented.

**Pull request merged**